tools/update_blueprint_needs.py

```diff
@@ -56,7 +56,7 @@ def main(argv):
 
     auto_update_blueprint = re.compile("~~~ auto-update-blueprint: (.+)")
     auto_update_blueprint_end = re.compile("~~~ end-auto-update")
-    json_filename = re.compile(".+\\.json")
+    json_filename = re.compile("[^.].+\\.json")
```

Comment:
```suggestion
    json_filename = re.compile("[^.].*\\.json")
```
In case any json file name contains only one character apart from the extension.